vecstore: init at 1.0.0

PhilipJohnBasile opened the pull request:
**Description:**

High-performance embeddable vector database with HNSW indexing, hybrid search, and comprehensive RAG toolkit.

**Features:**
- HNSW vector indexing
- Dense and sparse vector search
- Hybrid search with BM25
- Multi-language bindings (Rust, Python, JavaScript)
- Production-ready server mode

**Links:**
- Homepage: https://github.com/PhilipJohnBasile/vecstore
- Release: https://github.com/PhilipJohnBasile/vecstore/releases/tag/v1.0.0
- docs.rs: https://docs.rs/vecstore

**Testing:**
```bash
nix-build -A vecstore
./result/bin/vecstore-server --help
```

---
*Automated submission from GitHub Actions*

github-actions[bot] committed:
vecstore: init at 1.0.0

yzhou216 reviewed (changes requested):
Please format the file with the latest `nixfmt` and [add yourself as a maintainer first](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#how-to-become-a-maintainer).

PhilipJohnBasile committed:
vecstore: add philipjohnbasile as maintainer

- Added philipjohnbasile to maintainer-list.nix
- Updated package.nix to reference philipjohnbasile as maintainer

Note: nixfmt formatting will be applied separately once Nix is installed locally.

yzhou216 reviewed (changes requested):
The commit that adds yourself as a maintainer should be before the changes to other files, and the commit message should be `maintainers: add philipjohnbasile` instead.

qzylinra commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+  # Build only the server binary
+  cargoBuildFlags = [ "--bin" "vecstore-server" ];
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

qzylinra commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+  meta = with lib; {
+    description = "High-performance embeddable vector database (100/100 feature matrix vs competitors) - HNSW, hybrid search, multi-language, production-ready";
+    homepage = "https://github.com/PhilipJohnBasile/vecstore";
+    license = licenses.mit;
```
```suggestion
    license = lib.licenses.mit;
```

qzylinra commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+    protobuf
+  ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [
```
```suggestion
  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
```

FliegendeWurst commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+    owner = "PhilipJohnBasile";
+    repo = "vecstore";
+    rev = "v${version}";
+    hash = "sha256-REPLACE_WITH_ACTUAL_HASH";
+  };
+
+  cargoHash = "sha256-REPLACE_WITH_ACTUAL_CARGO_HASH";
```
You need to replace these with the right hash.

yzhou216 commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
```
Use `finalAttrs` pattern instead.

yzhou216 commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+    darwin.apple_sdk.frameworks.SystemConfiguration
+  ];
```
I'm not sure if this is still needed.

yzhou216 commented on pkgs/by-name/ve/vecstore/package.nix:
```diff
@@ -0,0 +1,46 @@
+
+  # Build only the server binary
+  cargoBuildFlags = [ "--bin" "vecstore-server" ];
+
```
Add the update script.

PhilipJohnBasile committed:
Add temporary VecStore Nixpkgs refresh workflow

PhilipJohnBasile committed:
Publish refreshed VecStore Nix package on a clean sync branch

PhilipJohnBasile committed:
Merge current Nixpkgs master and refresh VecStore package

PhilipJohnBasile committed:
Compute and validate VecStore cargo hash

PhilipJohnBasile committed:
Evaluate VecStore package through pinned remote Nixpkgs
